compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -18,13 +18,10 @@ fn resolve_instance<'tcx>(
 
     let result = if let Some(trait_def_id) = tcx.trait_of_item(def) {
         debug!(" => associated item, attempting to find impl in param_env {:#?}", param_env);
-        resolve_associated_item(
-            tcx,
-            def,
-            param_env,
-            trait_def_id,
-            tcx.normalize_erasing_regions(param_env, substs),
-        )
+        match tcx.try_normalize_erasing_regions(param_env, substs) {
```

Comment:
The normalization needs to be introduced somewhere higher in the call stack, not here. Normalizing on every call to `resolve_instance` is far too general, I think.